src/SixLabors.Core/Helpers/DebugGuard.cs

```diff
@@ -197,5 +217,45 @@ public static void MustBeSizedAtLeast<T>(ReadOnlySpan<T> target, ReadOnlySpan<T>
                 throw new ArgumentException($"Span-s must be at least of length {minSpan.Length}!", parameterName);
             }
         }
+
+        /// <summary>
+        /// Verifies, that the `target` span has the length of 'minSpan', or longer.
+        /// </summary>
+        /// <typeparam name="T">The element type of the spans</typeparam>
+        /// <param name="target">The target span.</param>
+        /// <param name="minSpan">The 'minSpan' span to compare 'target' to.</param>
+        /// <param name="parameterName">The name of the parameter that is to be checked.</param>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="target"/> is true
+        /// </exception>
+        [Conditional("DEBUG")]
+        public static void MustBeSizedAtLeast<T>(Span<T> target, Span<T> minSpan, string parameterName)
+            where T : struct
+        {
+            if (target.Length < minSpan.Length)
+            {
+                throw new ArgumentException($"Span-s must be at least of length {minSpan.Length}!", parameterName);
+            }
+        }
+
+        /// <summary>
+        /// Verifies, that the `target` array has declared the length  or longer.
+        /// </summary>
+        /// <typeparam name="T">The element type of the spans</typeparam>
+        /// <param name="target">The target array.</param>
+        /// <param name="minLength">The min length the array must have.</param>
+        /// <param name="parameterName">The name of the parameter that is to be checked.</param>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="target"/> is true
+        /// </exception>
+        [Conditional("DEBUG")]
+        public static void MustBeSizedAtLeast<T>(T[] target, int minLength, string parameterName)
```

Comment:
T[] -> Span<T> isn't free. I figured for guards we my as well have the overload, its not like the guard will likely change once its in.